### PR TITLE
add fibonacci functions

### PR DESCRIPTION
**Pull request by kontrybutor**

### CODE REVIEW

File: module1/homework/fibonacci/fibonacci.hpp [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
+
 int fibonacci_iterative(int sequence) {
-    // TODO: Your implementation goes here
+
+  if (sequence == 0) {
     return 0;
+  }
+  std::vector<int> v(sequence, 1);
+  std::adjacent_difference(v.begin(), v.end() - 1, v.begin() + 1,
+                           std::plus<int>{});
```
Comment (MechaniQ1) [inline module1/homework/fibonacci/fibonacci.hpp:14]: Nice idea, but you can also do that with 3 variables only. Less memory needed.

Comment (kontrybutor) [reply module1/homework/fibonacci/fibonacci.hpp:14]: Thanks, fully agree. I measured execution time and solution with for-loop and 3 variables is approx. 10 times faster than my solution.  Anyway, I would like to keep my solution as I promised myself to get to know as much of STL algorithms as possible.

Comment (nauka-programowania-MA) [reply module1/homework/fibonacci/fibonacci.hpp:14]: using STL is safe, because STL is already tested :)

**Comment by ziobron** [comment]
{ "requirement_ids": [1119, 1430], "points": [2, 6] }

**Pull request closed without merging**